lasagne/nonlinearities.py

```diff
@@ -269,6 +269,70 @@ def elu(x):
     return theano.tensor.switch(x > 0, x, theano.tensor.expm1(x))
 
 
+
+# selu
+class SELU(object):
+    """Scaled Exponential Linear Unit
+    :math:`\\varphi(x) = \\lambda (x > 0 ? x : \\alpha(e^x-1)`
+
+    The Scaled Exponential Linear Unit (SELU) was introduced in [1]_
+    as an activation function that allows the construction of
+    self-normalizing neural networks.
+
+    Parameters
+    ----------
+    scale : float32
+        The scale parameter :math:`\\lambda` for scaling all output.
+
+    scale_neg  : float32
+        The scale parameter :math:`\\alpha`
+        for scaling output for negative argument values.
+
+    Methods
+    -------
+    __call__(x)
+        Apply the SELU function to the activation `x`.
+
+    Examples
+    --------
+    In contrast to other activation functions in this module, this is
+    a class that needs to be instantiated to obtain a callable:
+
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> l_in = InputLayer((None, 100))
+    >>> from lasagne.nonlinearities import SELU
+    >>> selu = SELU(2, 3)
+    >>> l1 = DenseLayer(l_in, num_units=200, nonlinearity=selu)
+
+    See Also
+    --------
+    selu: Instance with :math:`\\alpha=1.6733, \\lambda=1.0507`, as recommended in [1]_.
+
+    References
+    ----------
+    .. [1] Günter Klambauer, Thomas Unterthiner, Andreas Mayr, Sepp Hochreiter (2017):
+       Self-Normalizing Neural Networks,
+       https://arxiv.org/abs/1706.02515
+    """
+
+    def __init__(self, scale=1, scale_neg=1):
+        self.scale = scale
+        self.scale_neg = scale_neg
+
+    def __call__(self, x):
+        return self.scale * theano.tensor.switch(
+            x > 0.0,
+            x,
+            self.scale_neg * (theano.tensor.exp(x) - 1))
+
+
+selu = SELU(scale=1.0507, scale_neg=1.6733)
+selu.__doc__ = """selu(x)
+    Instance of :class:`SELU` with :math:`\\alpha=1.6733, \\lambda=1.0507`
+    (the values that were recommended in the original paper or self-normalzing neworks)
```

Comment:
Should be two sentences, because the first sentence is used for the overview table at the beginning of the module documentation. I'd suggest:
```
Instance of :class:`SELU` with :math:`\\alpha \\approx 1.6733, \\lambda \\approx 1.0507`.
This has a stable and attracting fixed point of :math:`\\mu=0`, :math:`\\sigma=1` under
the assumptions of the original paper on self-normalizing neural networks.
```